lib/ansible/modules/network/f5/bigip_command.py

```diff
@@ -0,0 +1,379 @@
+    server: "lb.mydomain.com"
+    password: "secret"
+    user: "admin"
+    validate_certs: "no"
+  delegate_to: localhost
+'''
+
+RETURN = '''
+stdout:
+    description: The set of responses from the commands
+    returned: always
+    type: list
+    sample: ['...', '...']
+
+stdout_lines:
+    description: The value of stdout split into a list
+    returned: always
+    type: list
+    sample: [['...', '...'], ['...'], ['...']]
+
+failed_conditions:
+    description: The list of conditionals that have failed
+    returned: failed
+    type: list
+    sample: ['...', '...']
+'''
+
+import time
+
+from ansible.module_utils.f5_utils import AnsibleF5Client
+from ansible.module_utils.f5_utils import AnsibleF5Parameters
+from ansible.module_utils.f5_utils import HAS_F5SDK
+from ansible.module_utils.f5_utils import F5ModuleError
+from ansible.module_utils.f5_utils import iControlUnexpectedHTTPError
+from ansible.module_utils.netcli import FailedConditionsError
+from ansible.module_utils.six import string_types
+from ansible.module_utils.netcli import Conditional
+from ansible.module_utils.network_common import ComplexList
+from ansible.module_utils.network_common import to_list
+from collections import deque
+
+
+class Parameters(AnsibleF5Parameters):
+    returnables = ['stdout', 'stdout_lines', 'warnings']
+
+    def to_return(self):
+        result = {}
+        for returnable in self.returnables:
+            result[returnable] = getattr(self, returnable)
+        result = self._filter_params(result)
+        return result
+
+    @property
+    def commands(self):
+        commands = deque(self._values['commands'])
+        commands.appendleft(
+            'tmsh modify cli preference pager disabled'
+        )
+        commands = map(self._ensure_tmsh_prefix, list(commands))
+        return list(commands)
+
+    def _ensure_tmsh_prefix(self, cmd):
+        cmd = cmd.strip()
+        if cmd[0:5] != 'tmsh ':
+            cmd = 'tmsh ' + cmd.strip()
+        return cmd
+
+
+class ModuleManager(object):
+    def __init__(self, client):
+        self.client = client
+        self.want = Parameters(self.client.module.params)
+        self.changes = Parameters()
+
+    def _to_lines(self, stdout):
+        lines = list()
+        for item in stdout:
+            if isinstance(item, string_types):
+                item = str(item).split('\n')
+            lines.append(item)
+        return lines
+
+    def _is_valid_mode(self, cmd):
+        valid_configs = [
+            'tmsh list', 'tmsh show',
+            'tmsh modify cli preference pager disabled'
+        ]
+        if any(cmd.startswith(x) for x in valid_configs):
+            return True
+        return False
+
+    def exec_module(self):
+        result = dict()
+
+        try:
+            self.execute()
+        except iControlUnexpectedHTTPError as e:
+            raise F5ModuleError(str(e))
+
+        result.update(**self.changes.to_return())
+        result.update(dict(changed=False))
```

Comment:
this is far easier said that done as BIG-IP does not have a consistent output for when things change, nor does it even have consistent rc reporting. It's an unfortunate oversight on F5's part, but at this time I'll change this to `True` because there is no reliable way to get this information.